pkg/guard-gate/gate.go

```diff
@@ -0,0 +1,221 @@
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package guardgate
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	pi "knative.dev/security-guard/pkg/pluginterfaces"
+
+	utils "knative.dev/security-guard/pkg/guard-utils"
+)
+
+const plugVersion string = "0.0.1"
```

Comment:
ideally we could pick this dynamically rather than having to hardcore (and have to remember to update on each release)... pretty sure we do this elsewhere in knative, let me find an example...

Reply:
Think I was thinking of a combination of https://github.com/knative/pkg/pull/2548 and https://github.com/knative-sandbox/kn-plugin-quickstart/pull/327

Still wonder if there's a better way to get this info (maybe similar to what we do in serving to [add the version as a label](https://github.com/knative/serving/blob/646d0b1a9626cf636faf3e9b3a58bdd8eeb90ee5/hack/generate-yamls.sh#L77)), but it isn't blocking